cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -420,7 +420,12 @@ private DatasetTypeMeta getTypeInfo(Id.Namespace namespaceId, String typeName) t
       typeMeta = typeManager.getTypeInfo(systemDatasetTypeId);
     } else {
       // not using a system dataset type. ensure that the user has access to it before returning
-      ensureAccess(namespaceId.toEntityId().datasetType(typeName));
+      DatasetTypeId typeId = datasetTypeId.toEntityId();
+      Principal principal = authenticationContext.getPrincipal();
+      Predicate<EntityId> filter = authorizer.createFilter(principal);
```

Comment:
This is the temporary workaround. If you have an app that contains a custom dataset type (e.g. PurchaseStore), during deployment, after the dataset type gets created, you're granted ALL privileges on that app. However, this step currently does not invalidate the cache for the principal deploying the app. So this new grant is not in the cache. When the deployment pipeline now tries to create a dataset instance using this custom dataset, if you use the cache for enforcement, it will almost always fail. So just for this one particular case, adding a temporary workaround right now to bypass the cache. Need a better fix, but may not happen in 3.5